src/tests/Interop/COM/ExtensionPoints/ExtensionPoints.cs

```diff
@@ -0,0 +1,75 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.InteropServices;
+
+using COM;
+
+using TestLibrary;
+using Xunit;
+
+public class ExtensionPoints
+{
+    unsafe class MallocSpy : IMallocSpy
+    {
+        private int _called = 0;
+        public int Called => _called;
+
+        public virtual nuint PreAlloc(nuint cbRequest)
+        {
+            _called++;
+            return cbRequest;
+        }
+
+        public virtual unsafe void* PostAlloc(void* pActual) => pActual;
+        public virtual unsafe void* PreFree(void* pRequest, [MarshalAs(UnmanagedType.Bool)] bool fSpyed)
+        {
+            _called++;
+            return pRequest;
+        }
+
+        public virtual void PostFree([MarshalAs(UnmanagedType.Bool)] bool fSpyed) { }
+        public virtual unsafe nuint PreRealloc(void* pRequest, nuint cbRequest, void** ppNewRequest, [MarshalAs(UnmanagedType.Bool)] bool fSpyed) => cbRequest;
+        public virtual unsafe void* PostRealloc(void* pActual, [MarshalAs(UnmanagedType.Bool)] bool fSpyed) => pActual;
+        public virtual unsafe void* PreGetSize(void* pRequest, [MarshalAs(UnmanagedType.Bool)] bool fSpyed) => pRequest;
+        public virtual nuint PostGetSize(nuint cbActual, [MarshalAs(UnmanagedType.Bool)] bool fSpyed) => cbActual;
+        public virtual unsafe void* PreDidAlloc(void* pRequest, [MarshalAs(UnmanagedType.Bool)] bool fSpyed) => pRequest;
+        public virtual unsafe int PostDidAlloc(void* pRequest, [MarshalAs(UnmanagedType.Bool)] bool fSpyed, int fActual) => fActual;
+        public virtual void PreHeapMinimize() { }
+        public virtual void PostHeapMinimize() { }
+    }
+
+    [Fact]
+    public static unsafe void Validate_Managed_IMallocSpy()
+    {
+        Console.WriteLine($"Running {nameof(Validate_Managed_IMallocSpy)}...");
+        var mallocSpy = new MallocSpy();
+        int result = Ole32.CoRegisterMallocSpy(mallocSpy);
+        Assert.Equal(0, result);
+        try
+        {
+            var arr = new [] { "", "", "", "", null };
+
+            var fptr = (delegate*unmanaged<string[], int>)(delegate*unmanaged<char**, int>)&ArrayLen;
```

Comment:
I'd like us to add a comment here explaining exactly why we're casting here and what is actually under test (that the runtime-marshalling system correctly switches GC modes before calling `CoTaskMemAlloc` and family when using strings and arrays). I know what this is doing now, but I'm pretty sure we'll forget soon and not know what we're specifically testing here.